multicluster/controllers/multicluster/commonarea/labelidentityimport_controller.go

```diff
@@ -0,0 +1,166 @@
+	}
+	return r.handleLabelIdentityImpUpdate(ctx, &labelIdentityImport)
+}
+
+func (r *LabelIdentityImportReconciler) handleLabelIdentityImpUpdate(ctx context.Context,
+	labelImp *multiclusterv1alpha1.LabelIdentityImport) (ctrl.Result, error) {
+	labelIdentityName := types.NamespacedName{
+		Name: labelImp.Name,
+	}
+	labelIdentity := &multiclusterv1alpha1.LabelIdentity{}
+	err := r.localClusterClient.Get(ctx, labelIdentityName, labelIdentity)
+	labelIdentityNotFound := apierrors.IsNotFound(err)
+	if err != nil && !labelIdentityNotFound {
+		return ctrl.Result{}, err
+	}
+	if labelIdentityNotFound {
+		newLabelIdentity := &multiclusterv1alpha1.LabelIdentity{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: labelImp.Name,
+			},
+			Spec: multiclusterv1alpha1.LabelIdentityImportSpec{
+				Label: labelImp.Spec.Label,
+				ID:    labelImp.Spec.ID,
+			},
+		}
+		if err = r.localClusterClient.Create(ctx, newLabelIdentity, &client.CreateOptions{}); err != nil {
+			klog.ErrorS(err, "Failed to create LabelIdentity", "clusterID", r.localClusterID, "label", labelImp.Spec.Label)
+			return ctrl.Result{}, err
+		}
+		r.installedLabelImports.Add(*labelImp)
+	} else {
+		// ID assigned for LabelIdentity could change in case of leader controller restart
+		if labelIdentity.Spec.ID != labelImp.Spec.ID {
+			labelIdentity.Spec.ID = labelImp.Spec.ID
+			if err = r.localClusterClient.Update(ctx, labelIdentity, &client.UpdateOptions{}); err != nil {
+				klog.ErrorS(err, "Failed to update LabelIdentity", "clusterID", r.localClusterID, "label", labelImp.Spec.Label)
+				return ctrl.Result{}, err
+			}
+		}
+	}
+	return ctrl.Result{}, nil
+}
+
+func (r *LabelIdentityImportReconciler) handleLabelIdentityImpDelete(ctx context.Context,
+	labelImp *multiclusterv1alpha1.LabelIdentityImport) (ctrl.Result, error) {
+	labelIdentityName := types.NamespacedName{
+		Name: labelImp.Name,
+	}
+	klog.V(2).InfoS("Deleting LabelIdentity corresponding to LabelIdentityImport", "labelidentityimport", klog.KObj(labelImp))
+	labelIdentity := &multiclusterv1alpha1.LabelIdentity{}
+	err := r.localClusterClient.Get(ctx, labelIdentityName, labelIdentity)
+	if err != nil {
+		klog.V(2).InfoS("Unable to fetch LabelIdentity", "labelidentity", labelIdentityName.Name, "err", err)
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	err = r.localClusterClient.Delete(ctx, labelIdentity, &client.DeleteOptions{})
+	if err != nil {
+		klog.ErrorS(err, "Failed to delete LabelIdentity", "labelidentity", labelIdentityName.Name)
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	r.installedLabelImports.Delete(*labelImp)
+	return ctrl.Result{}, nil
+}
+
+func (r *LabelIdentityImportReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// Ignore status update event via GenerationChangedPredicate
+	instance := predicate.GenerationChangedPredicate{}
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&multiclusterv1alpha1.LabelIdentityImport{}).
```

Comment:
Could you check if it's OK to wrap `LabelIdentityImport` into `ResourceImport`? I believe it should be similar like `ServiceImport`.

Reply:
Done.